homeassistant/components/sensor/__init__.py

```diff
@@ -531,7 +542,14 @@ def unit_of_measurement(self) -> str | None:
         ):
             return self.hass.config.units.temperature_unit
 
-        # Fourth priority: Native unit
+        # Fourth priority: Unit translation
+        if (translation_key := self._unit_of_measurement_translation_key) and (
```

Comment:
Should we blow up if the translation was found, but a native unit of measurement was set as well?